b2share/modules/records/serializers/schemas/json.py

```diff
@@ -43,19 +47,35 @@ class DraftSchemaJSONV1(Schema):
     @pre_dump
     def filter_internal(self, data):
         """Remove internal fields from the record metadata."""
+        external_pids = []
+        bucket = None
+        try:
+            record = g.record
+            external_pids = generate_external_pids(record)
```

Comment:
There is no need to generate the list of external_pids for published records as their files won't change.